### PR TITLE
Lesson16

### PR DESCRIPTION
**Pull request by MaxFedosenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    <artifactId>JavaDev</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <name>MavenProject</name>
+    <!-- FIXME change it to the project's website -->
+    <url>http://www.example.com</url>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>1.7</maven.compiler.source>
```
Comment (gr00shik) [inline pom.xml]: set 1.8

File: src/com/tms/lesson15/persistance/CarService.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import com.tms.lesson15.entity.TYPE;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CarService implements Entity<Car> {
+
+    private Connection connect;
```
Comment (gr00shik) [inline src/com/tms/lesson15/persistance/CarService.java:13]: please make sure that connection is not null

File: src/com/tms/lesson15/persistance/CarService.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        }
+        return false;
+    }
+
+    @Override
+    public boolean delete(Car car) {
+        try {
+            int id = car.getId();
+            if (id == 0) {
+                throw new SQLException("no car in database");
```
Comment (gr00shik) [inline src/com/tms/lesson15/persistance/CarService.java]: you don't check database, you just validate that car doesn't contain id

Comment (gr00shik) [reply src/com/tms/lesson15/persistance/CarService.java]: possible solution: return message - car must contain id

**Review by gr00shik — COMMENTED** [review]
ok

File: src/com/tms/lesson15/persistance/CarService.java (hunk 2 above)
Comment (MaxFedosenko) [reply src/com/tms/lesson15/persistance/CarService.java:13]: не совсем понимаю где я должен сделать проверку на null...

Comment (gr00shik) [reply src/com/tms/lesson15/persistance/CarService.java:13]: throw exception from configuration class

Comment (MaxFedosenko) [reply src/com/tms/lesson15/persistance/CarService.java:13]: I understand, thank you

**Pull request merged**